ng/src/web/pages/groups/dialog.js

```diff
@@ -30,103 +31,117 @@ import Layout from '../../components/layout/layout.js';
+
+  const defaults = {name: _('Unnamed'), users: []};
 
-  const is_edit = is_defined(id);
   return (
-    <Layout flex="column">
-      <FormGroup title={_('Name')}>
-        <TextField
-          name="name"
-          grow="1"
-          value={name}
-          size="30"
-          onChange={onValueChange}
-          maxLength="80"/>
-      </FormGroup>
-
-      <FormGroup
-        title={_('Comment')}
-        flex="column">
-        <TextField
-          name="comment"
-          value={comment}
-          size="30"
-          maxLength="400"
-          onChange={onValueChange}/>
-      </FormGroup>
-
-      <FormGroup
-        title={_('Users')}>
-        <MultiSelect
-          name="users"
-          value={users}
-          onChange={onValueChange}
-        >
-          {
-            map(all_users, user => {
-              return (
-                <option
-                  key={user.id}
-                  value={user.name}>
-                  {user.name}
-                </option>
-              );
-            })
-          }
-        </MultiSelect>
-      </FormGroup>
-
-      {!is_edit &&
-        <FormGroup
-          title={_('Special Groups')}>
-          <Checkbox
-            name="grant_full"
-            checkedValue="1"
-            unCheckedValue="0"
-            checked={grant_full === '1'}
-            title={_('Create permission to grant full read and write access ' +
-              'among all group members and across any resources')}
-            onChange={onValueChange}
-          />
-        </FormGroup>
-      }
-    </Layout>
+    <SaveDialog
+      visible={visible}
+      title={title}
+      onClose={onClose}
+      onSave={onSave}
+      initialData={{
+        ...defaults,
+        ...group,
+      }}
+    >
+      {({
+        data: state,
+        onValueChange,
+      }) => {
+        return (
+          <Layout flex="column">
+            <FormGroup title={_('Name')}>
+              <TextField
+                name="name"
+                grow="1"
+                value={state.name}
+                size="30"
+                onChange={onValueChange}
+                maxLength="80"/>
+            </FormGroup>
+
+            <FormGroup
+              title={_('Comment')}
+              flex="column">
+              <TextField
+                name="comment"
+                value={state.comment}
+                size="30"
+                maxLength="400"
+                onChange={onValueChange}/>
+            </FormGroup>
+
+            <FormGroup
+              title={_('Users')}>
+              <MultiSelect
+                name="users"
+                value={state.users}
+                onChange={onValueChange}
+              >
+                {
+                  map(allUsers, user => {
+                    return (
+                      <option
+                        key={user.id}
+                        value={user.name}>
+                        {user.name}
+                      </option>
```

Comment:
Please create an items array instead of option elements. option elements will be converted into items internally. They are only provided for backwards compatibility.